near/omni-bridge/src/lib.rs

```diff
@@ -1141,6 +1320,22 @@ impl Contract {
             .saturating_mul((env::storage_usage().saturating_sub(storage_usage)).into())
     }
 
+    fn add_fast_transfer(
+        &mut self,
+        fast_transfer: &FastTransfer,
+        relayer_id: &AccountId,
+    ) -> NearToken {
+        let storage_usage = env::storage_usage();
+        require!(
+            self.fast_transfers
+                .insert(&fast_transfer.id(), relayer_id)
+                .is_none(),
+            "Fast transfer is already performed"
```

Comment:
We can check it but we can't insert a fast transfer before storage deposit. So the race condition can still happen if two relayers submit fast transfer in the same block. @karim-en do we need to implement refund logic?

Reply:
It doesn't make much sense to me. Even if we checked it, it is possible that this could be failed in the next block.
But I'm a little bit surprised that this implementation works, because the last promise `send_tokens` should return value, so the `resolve_transfer` will be able to process refund for the failed `ft_on_transfer`.

Reply:
I think I understood what you meant @karim-en. Is this a fix f046b2d5fc0efa9bb7df1ea1837dc05b93122919?

Reply:
Yeh, it looks good now.
But the naming could be confusing (in the explorer) because the nep141 implementation also has a `ft_resolve_transfer`

Reply:
Renamed 1f38b0469619ad975744e128cc91dc404cde9b6f